<launch>
    <arg name="namespace_prefix" default="mantra_gazebo" />

    <arg name="markerId"        default="100"/>
    <arg name="markerSize"      default="0.1"/>    <!-- in m -->
    <arg name="marker_frame"    default="aruco_marker"/>
    <arg name="ref_frame"       default=""/>  <!-- leave empty and the pose will be published wrt param parent_name -->
    <arg name="corner_refinement" default="LINES" /> <!-- NONE, HARRIS, LINES, SUBPIX -->

    <!-- start ArUco -->
    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="/camera/color/camera_info" />
        <remap from="/image" to="/camera/color/image_raw" />
        <param name="image_is_rectified" value="false"/> <!-- use raw image -->
        <param name="marker_size"        value="$(arg markerSize)"/>
        <param name="marker_id"          value="$(arg markerId)"/>
        <param name="reference_frame"    value="$(arg ref_frame)"/>   <!-- frame in which the marker pose will be refered -->
        <param name="camera_frame"       value="camera_link_optical"/>
        <param name="marker_frame"       value="$(arg marker_frame)" />
        <param name="corner_refinement"  value="$(arg corner_refinement)" />
    </node>

    <!-- start the robot -->
    <include file="$(find mantra_gazebo)/launch/mantra_moveit_gazebo.launch" >
        <arg name="use_rviz" value="false" />
    </include>

    <!-- robot hmi -->
    <include file="$(find mantra_hmi)/launch/mantra_hmi.launch" />

    <!-- move server -->
    <node name="mantra_move_server" pkg="mantra_application" type="mantra_move_server.py" respawn="true" output="screen" />

    <!-- start easy_handeye -->
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="true" />
        <arg name="publish_dummy" value="true" />

        <arg name="tracking_base_frame" value="camera_link_optical" />
        <arg name="tracking_marker_frame" value="aruco_marker" />
        <arg name="robot_base_frame" value="base_link" />
        <arg name="robot_effector_frame" value="ee_link" />
        <arg name="move_group" value="arm" />

        <arg name="rviz_config_file" value="$(find mantra_application)/calibration/config/calibration.rviz" />
        <arg name="rqt_perspective_file" value="$(find mantra_application)/calibration/config/calibration.perspective" />

        <arg name="freehand_robot_movement" value="true" />
        <arg name="robot_velocity_scaling" value="0.5" />
        <arg name="robot_acceleration_scaling" value="0.2" />
    </include>

    <!-- t_e_c
    - Translation: [-0.070, 0.000, 0.0345]
    - Rotation: in Quaternion [0.000, -0.000, -0.707, 0.707]
                in RPY (radian) [0.000, -0.000, -1.571]
                in RPY (degree) [0.000, -0.000, -90.000] -->

    <!-- marker -->
    <!-- [0.45, 0, 0.002] [pi/2, 0, pi/2]-->

</launch>